File: Schoolscape_2/Part_2/Yard.java
package myschool;

import static myschool.Configuration.MAX_ARRAY_SIZE;
import static myschool.Configuration.locationOut;
import static myschool.Configuration.locationYard;

public class Yard extends Place {

    //ARRAY CONTAINING STUDENTS IN THE YARD
    private Student[] studentsIn;
    //NUMBER OF STUDENT IN THE YARD
    private int numberOfStudentsIn;
    //YARD CAPACITY
    private int capacity;

    public Yard() {
        int i;
        this.numberOfStudentsIn = 0;
        this.capacity = 0;
        this.type = "yard";
        System.out.println("Yard constructed");

    }

    public Yard(int c) {
        int i;
        this.numberOfStudentsIn = 0;
        this.capacity = c;
        this.type = "yard";
        this.studentsIn = new Student[MAX_ARRAY_SIZE];
        for (i = 0; i < MAX_ARRAY_SIZE; i++) {
            this.studentsIn[i] = null;
        }
        System.out.println("Yard constructed");

    }

    public int addStudent(Student s) {
        if (this.numberOfStudentsIn < this.capacity) {//IF THERE IS ENOUGH SPACE
            //Entering school
            System.out.println(s.getFullName() + " enters school.");
            //Entering yard
            System.out.println(s.getFullName() + " enters schoolyard.");
            s.setLocation(locationYard);
            this.studentsIn[this.numberOfStudentsIn++] = s;
            return 1;
        }
        return 0;
    }

    public int addStudent(Student s, int x) {
        if (this.numberOfStudentsIn < this.capacity) {//IF THERE IS ENOUGH SPACE
            //Entering yard
            System.out.println(s.getFullName() + " enters schoolyard.");
            s.setLocation(locationYard);
            this.studentsIn[this.numberOfStudentsIn++] = s;
            return 1;
        }
        return 0;
    }

    public void print() {
        System.out.println("People in yard are:");
        int i;
        for (i = 0; i < this.numberOfStudentsIn; i++) {
            studentsIn[i].print();
        }
    }

    public int getCapacity() {
        return this.capacity;
    }

    public void setCapacity(int c) {
        this.capacity = c;
    }

    public int addStudentToStairs(Stairs st, Student s) {
        int i;
        //IF STUDENT IS NOT IN THE YARD DO NOTHING
        for (i = 0; i < this.numberOfStudentsIn; i++) {
            if (this.studentsIn[i].getId() == s.getId()) {
                break;
            }
        }
        if (i >= this.numberOfStudentsIn) {
            return 0;
        }
        //ADDS TO STAIRS
        if (st.addStudent(s) == 1) {

            //EXITS - REPLACE CURRENT STUDENT WITH THE LAST IN THE ARRAY
            this.studentsIn[i] = this.studentsIn[this.numberOfStudentsIn - 1];
            this.studentsIn[this.numberOfStudentsIn - 1] = null;
            this.numberOfStudentsIn--;
        } else {
            return 0;
        }
        return 1;
    }

    public Student exit() {
        if (this.numberOfStudentsIn > 0) {
            Student s = studentsIn[--numberOfStudentsIn];
            System.out.println(s.getFullName() + " exits yard!");
            System.out.println(s.getFullName() + " exits building!");
            s.setLocation(locationOut);
            return s;
        }
        return null;
    }
}

File: Schoolscape_2/Part_1/Yard.cpp
#include "Yard.h"

		Yard::Yard(){
			//int i;
			this->numberOfStudentsIn = 0;
			this->capacity = 0;
			this->type = "yard";
			cout << "Yard constructed" << endl;
		}
		Yard::Yard(int c){
			int i;
			this->numberOfStudentsIn = 0;
			this->capacity = c;
			this->type = "yard";
			for (i=0;i<MAX_ARRAY_SIZE;i++){
				this->studentsIn[i] = NULL;
			}
			cout << "Yard constructed" << endl;
		}
		Yard::~Yard(){
			int i;
			for (i=0;i<MAX_ARRAY_SIZE;i++){
				if (this->studentsIn[i]!=NULL){
					delete(this->studentsIn[i]);
				}
			}
			cout << "Corridor Destroyed" << endl;
		}
		int Yard::addStudent(Student *s){
			if (this->numberOfStudentsIn<this->capacity){//IF THERE IS ENOUGH SPACE
				//Entering school
				cout <<  s->getFullName() << " enters school."<<endl<<endl;
				//Entering yard
				cout << s->getFullName() << " enters schoolyard." << endl<<endl;
				s->setLocation(locationYard);
				this->studentsIn[this->numberOfStudentsIn++] = s;
				return 1;
			}
			return 0;
		} 
		
		void Yard::print(){
			cout << "People in yard are:" << endl;
			int i;
			for (i=0;i<this->numberOfStudentsIn;i++){
				studentsIn[i]->print();
			}
		}
		
		int Yard::getCapacity(){
			return this->capacity;
		}
		
		void Yard::setCapacity(int c){
			this->capacity = c;
		}
		
		int Yard::addStudentToStairs(Stairs *st,Student *s){
			int i;
			//IF STUDENT IS NOT IN THE YARD DO NOTHING
			for (i=0;i<this->numberOfStudentsIn;i++){
				if (this->studentsIn[i]->getId()==s->getId()){
					break;
				}
			}
			if (i>=this->numberOfStudentsIn){
				return 0;
			}
			//ADDS TO STAIRS
			if (st->addStudent(s)){
				
				//EXITS - REPLACE CURRENT STUDENT WITH THE LAST IN THE ARRAY
				this->studentsIn[i] = this->studentsIn[this->numberOfStudentsIn-1];
				this->studentsIn[this->numberOfStudentsIn-1] = NULL;
				this->numberOfStudentsIn--;
			}
			else {
				return 0;
			}
			return 1;
		}
